packages/cdktf/lib/backends/local-backend.ts

```diff
@@ -1,14 +1,24 @@
+import * as path from "path";
 import { Construct } from "constructs";
 import { TerraformBackend } from "../terraform-backend";
 import { keysToSnakeCase } from "../util";
 import {
   TerraformRemoteState,
   DataTerraformRemoteStateConfig,
 } from "../terraform-remote-state";
+import { TerraformStack } from "..";
 
 export class LocalBackend extends TerraformBackend {
-  constructor(scope: Construct, private readonly props: LocalBackendProps) {
+  private readonly props: LocalBackendProps;
+  constructor(scope: Construct, props: LocalBackendProps = {}) {
     super(scope, "backend", "local");
+
+    const stackId = TerraformStack.of(this).node.id;
+    this.props = {
+      ...props,
+      path:
+        props.path || path.join(process.cwd(), `terraform.${stackId}.tfstate`),
```

Comment:
Nice way of solving it 👍

This _could_ be dangerous, however, if users expect the Terraform default (`terraform.tfstate`) when not passing a path. We should at least document this somewhere. Maybe in a docstring for `LocalBackendProps > path`?